src/op_analytics/cli/subcommands/pulls/defillama/defillama_historical_chain_tvl.py

```diff
@@ -0,0 +1,201 @@
+from dataclasses import dataclass
+import polars as pl
+from op_coreutils.bigquery.write import (
+    most_recent_dates,
+    upsert_unpartitioned_table,
+)
+from op_coreutils.logger import structlog
+from op_coreutils.request import get_data, new_session
+from op_coreutils.threads import run_concurrently
+from op_coreutils.time import dt_fromepoch
+
+log = structlog.get_logger()
+
+CHAINS_METADATA_ENDPOINT = "https://api.llama.fi/config"
+CHAINS_ENDPOINT = "https://api.llama.fi/v2/chains"
+CHAINS_TVL_ENDPOINT = "https://api.llama.fi/v2/historicalChainTvl/{slug}"
+
+BQ_DATASET = "temp"
+
+CHAIN_METADATA_TABLE = "defillama_chains_metadata"
+HISTORICAL_CHAIN_TVL_TABLE = "defillama_daily_historical_chain_tvl"
+TVL_TABLE_LAST_N_DAYS = 100_000  # upsert only the last 7 days of tvl records fetched from the api
```

Comment:
here is where I think we need to agree on what the incremental updates approach will be.  Do we need to re-upsert the last 100k days every time?

Reply:
This is part of the "Do Not Merge" part as I'm still using this for testing purposes. I'll change it back to 7 days and change BQ_DATASET to "api_uploads".  I also need to flip the create new table flags to False

Reply:
sounds good! my recommendation while you are testing would be to comment out the upsert and replace with an overwrite call altogether.  This includes commenting out the part where the last N days are selected from the data.  

Otherwise the upsert utility will get in your way if you are trying to update too many partitions at once.